lib/streamlit/testing/v1/app_test.py

```diff
@@ -0,0 +1,256 @@
+
+        Useful for testing short scripts that fit comfortably as an inline
+        string in the test itself, without having to create a separate file
+        for it.
+
+        `default_timeout` is the default time in seconds before a script is
+        timed out, if not overridden for an individual `.run()` call.
+        """
+        hasher = hashlib.md5(bytes(script, "utf-8"))
+        script_name = hasher.hexdigest()
+
+        path = pathlib.Path(TMP_DIR.name, script_name)
+        aligned_script = textwrap.dedent(script)
+        path.write_text(aligned_script)
+        return AppTest(str(path), default_timeout=default_timeout)
+
+    @classmethod
+    def from_file(cls, script_path: str, default_timeout: float = 3) -> AppTest:
+        """Create a runner for the script with the given name, for testing.
+
+        `default_timeout` is the default time in seconds before a script is
+        timed out, if not overridden for an individual `.run()` call.
+        """
+        stack = traceback.StackSummary.extract(traceback.walk_stack(None))
+        filepath = pathlib.Path(stack[1].filename)
+        full_path = filepath.parent / script_path
+        return AppTest(str(full_path), default_timeout=default_timeout)
+
+    def _run(
+        self,
+        widget_state: WidgetStates | None = None,
+        timeout: float | None = None,
+    ) -> AppTest:
+        """Run the script, and parse the output messages for querying
+        and interaction.
+
+        Timeout is in seconds, or None to use the default timeout of the runner.
+        """
+        if timeout is None:
+            timeout = self.default_timeout
+
+        # setup
+        mock_runtime = MagicMock(spec=Runtime)
+        mock_runtime.media_file_mgr = MediaFileManager(
+            MemoryMediaFileStorage("/mock/media")
+        )
+        mock_runtime.cache_storage_manager = MemoryCacheStorageManager()
+        Runtime._instance = mock_runtime
+        with source_util._pages_cache_lock:
+            self.saved_cached_pages = source_util._cached_pages
+            source_util._cached_pages = None
+
+        with patch_config_options({"runner.postScriptGC": False}):
+            script_runner = LocalScriptRunner(self._script_path, self.session_state)
+            self._tree = script_runner.run(widget_state, self.query_params, timeout)
+            self._tree._runner = self
+
+        # teardown
+        with source_util._pages_cache_lock:
+            source_util._cached_pages = self.saved_cached_pages
+        Runtime._instance = None
+
+        return self
+
+    def run(self, timeout: float | None = None) -> AppTest:
+        """Run the script, and parse the output messages for querying
+        and interaction.
+
+        Timeout is in seconds, or None to use the default timeout of the runner.
+        """
+        return self._tree.run(timeout)
+
+    @property
+    def main(self) -> Block:
+        return self._tree.main
```

Comment:
I noticed that equivalent property for `sidebar` is missing on AppTest currently